R/check_tbl_values_required.R

```diff
@@ -5,18 +5,23 @@
 #' @inherit check_tbl_colnames params
 #' @inherit check_tbl_col_types return
 #' @export
-check_tbl_values_required <- function(tbl, round_id, file_path, hub_path) {
+check_tbl_values_required <- function(tbl, round_id, file_path, hub_path,
+                                      derived_task_ids = NULL) {
```

Comment:
Unfortunately, this function cannot be batched by output type. Rather it needs to continue to be batched by model task to ensure that if optional task ID values are supplied for a given model task, the correct required output types are also supplied. As such, we need to evaluate model tasks as a whole rather than splitting them unless we complicate this already very complicated function even more. I experimented a bit but decided it wasn't worth the effort at this time. Being able to ignore derived task ids already confers speed ups and I'm hoping I'll be able to gain more through speeding up `conc_rows()` in the next round of optimisations.